src/org/jitsi/impl/neomedia/RTPEncodingImpl.java

```diff
@@ -0,0 +1,413 @@
+    @Override
+    public MediaStreamTrackImpl getMediaStreamTrack()
+    {
+        return track;
+    }
+
+    /**
+     * Gets the subjective quality index of this instance.
+     *
+     * @return the subjective quality index of this instance.
+     */
+    int getIndex()
+    {
+        return subjectiveQualityIdx;
+    }
+
+    /**
+     * Returns a boolean that indicates whether or not this
+     * {@link RTPEncodingImpl} depends on the subjective quality index that is
+     * passed as an argument.
+     *
+     * @param idx
+     * @return true if this {@link RTPEncodingImpl} depends on the subjective
+     * quality index that is passed as an argument, false otherwise.
+     */
+    boolean requires(int idx)
+    {
+        if (idx < 0)
+        {
+            return false;
+        }
+
+        if (idx == this.subjectiveQualityIdx)
+        {
+            return true;
+        }
+
+        Boolean requires = dependencyCache.get(idx);
+
+        if (requires == null)
+        {
+            requires = false;
+
+            if (!ArrayUtils.isNullOrEmpty(dependencyEncodings))
+            {
+                for (RTPEncodingImpl enc : dependencyEncodings)
+                {
+                    if (enc.requires(idx))
+                    {
+                        requires = true;
+                        break;
+                    }
+                }
+            }
+
+            dependencyCache.put(idx, requires);
+        }
+
+        return requires;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean matches(byte[] buf, int off, int len)
+    {
+        long ssrc = RawPacket.getSSRCAsLong(buf, off, len);
+
+        if (primarySSRC != ssrc && rtxSSRC != ssrc)
+        {
+            return false;
+        }
+
+        int tid = track.getMediaStreamTrackReceiver()
+            .getStream().getTemporalID(buf, off, len);
+
+        return (tid == -1 && ArrayUtils.isNullOrEmpty(dependencyEncodings))
+                || tid == temporalId;
+    }
+
+    /**
+     * Gets a boolean flag that indicates whether or not this instance is
+     * streaming or if it's suspended.
+     *
+     * @param active true if this {@link RTPEncodingImpl} is active, otherwise
+     * false.
+     */
+    void setActive(boolean active)
+    {
+        this.active = active;
+    }
+
+    /**
+     *
+     * @param pkt
+     * @param nowMs
+     *
+     * @return the {@link SourceFrameDesc} that was updated, otherwise null.
+     */
+    SourceFrameDesc update(RawPacket pkt, long nowMs)
+    {
+        // Update rate stats.
+        rateStatistics.update(pkt.getLength(), nowMs);
+
+        long ts = pkt.getTimestamp();
+        SourceFrameDesc frame = frames.get(ts);
+
+        boolean seen = frame != null;
+        if (!seen)
+        {
+            frames.put(ts, frame = new SourceFrameDesc(this));
+
+            // We measure the stable bitrate on every new frame.
+            lastStableBitrateBps = getBitrateBps(nowMs);
```

Comment:
That's going to change soon, as we need this measurement in the JVB. Please let me know if you would like me to remove the code until the related JVB code lands in the bridge.